proto/api/v1/workspace_setting_service.proto

```diff
@@ -50,6 +50,11 @@ message WorkspaceGeneralSetting {
   // 0: Sunday, 1: Monday, 2: Tuesday, 3: Wednesday, 4: Thursday, 5: Friday, 6: Saturday
   // Default is Sunday.
   int32 week_start_day_offset = 6;
+
+  // disallow_change_username disallows changing username.
+  bool disallow_change_username = 7;
+  // disallow_change_nickname disallows changing nickname.
+  bool disallow_change_nickname = 8;
```

Comment:
Disabling username change is a nice feature, but the nickname part is a bit over-designed. And I haven't seen a platform that restricts changing nicknames. Is there more context behind this?

Reply:
You are right, restricting nickname is always a bad idea. However this option is made for some restrictive environment, like in my university (Peking Univ.), government regulations forces everyone to use their real name as nickname to avoid malformed nicknames.

Reply:
So I think it's reasonable to leave this options to the user ;D